src/particle.f90

```diff
@@ -0,0 +1,284 @@
+      ! write(*,*) '      time partical_swap :',real(part_comm_time,4)
+      ! write(*,*) '           alltoall comm :',real(a2a_time,4)
+      ! write(*,*) '           counting time :',real(count_time,4)
+      ! write(*,*) '           table shareing:',real(table_share_time,4)
+      ! write(*,*) '           data packing  :',real(data_pack_time,4)
+      ! write(*,*) '           MPI Alltoall  :',real(mpi_comm_time,4)
+      ! write(*,*) '           data unpacking:',real(data_unpack_time,4)
+      ! write(*,*) '                  hdf5 io:',real(h5io_time,4)
+    endif
+    !
+  end subroutine partcle_report
+  !+-------------------------------------------------------------------+
+  !| The end of the subroutine partcle_report.                         |
+  !+-------------------------------------------------------------------+
+
+  !+-------------------------------------------------------------------+
+  !| This subroutine is to initilise particle module.                  |
+  !+-------------------------------------------------------------------+
+  !| CHANGE RECORD                                                     |
+  !| -------------                                                     |
+  !| 17-11-2021  | Created by J. Fang                                  |
+  !+-------------------------------------------------------------------+
+  subroutine particle_init
+    !
+    use param,     only : xlx,yly,zlz,irestart,t,dt
+    use var,       only : itime
+    !
+    ! local data
+    ! integer :: i,j,k,p
+    ! real(mytype) :: dx,dy,dz
+    ! !
+    integer :: num_particle,n
+    real(mytype),allocatable,dimension(:) :: x_p
+
+    num_particle=100*(nrank+1)
+
+    allocate(x_p(num_particle))
+
+    do n=1,num_particle
+      call random_number(x_p(n))
+      x_p(n)=x_p(n)+nrank*100._mytype
+
+      ! print*,nrank,'|',n,'-',x_p(n)
+    enddo
+    
+    call pwrite('particle_init.bin',x_p)
+
+    ! if(irestart==0) then
+    !   !
+    !   call particle_gen(particle,numparticle)
+    !   !
+    !   particle_file_numb=0
+    !   !
+    !   call h5write_particle()
+    !   !
+    ! else
+    !   ! call h5read_particle(particle,numparticle)
+    !   call particle_gen(particle,numparticle)
+    !   !
+    !   particle_file_numb=0
+    !   !
+    !   call h5write_particle()
+    !   !
+    ! endif
+    ! !
+    ! ! call partical_domain_check('bc_tgv')
+    ! ! call partical_domain_check('bc_channel')
+    ! call partical_domain_check('out_disappear')
+    ! !
+    ! !
+    ! ! call partical_swap
+    ! !
+    ! ! call write_particle()
+    ! !
+    ! part_time=0.d0
+    ! part_comm_time=0.d0
+    ! part_vel_time=0.d0
+    ! part_dmck_time=0.d0
+    ! a2a_time=0.d0
+    ! count_time=0.d0
+    ! table_share_time=0.d0
+    ! data_pack_time=0.d0
+    ! data_unpack_time=0.d0
+    ! mpi_comm_time=0.d0
+    ! h5io_time=0.d0
+    ! !
+    ! particletime=t
+    ! sub_time_step=dt
+    ! sub_time_step=0.1d0*dt
+    !
+    ! lorentzforce=.true.
+    !
```

Comment:
Is this code no longer needed?

Reply:
I clicked the pull request by mistake... This part is under development. 
I will launch a pull request when it is done.
Just ignore or reject.